api/job-batches/params/process-batch-action-parameters.ts

```diff
@@ -1,7 +1,7 @@
-import { BaseParameters } from "../../parameters/index";
+import { BaseParameters } from "../../parameters";
 
 export class ProcessBatchActionParameters extends BaseParameters {
-    setFileUri(fileUri: string): ProcessBatchActionParameters {
+    setFileUri(fileUri: string): this {
```

Comment:
Consider someone for some reason would want to do a `setReason()` call after `setFileUri()`
![image](https://github.com/user-attachments/assets/d7dced15-caa3-488f-8966-3e78ba32cdb2)

Reply:
But `setReason` exists only for `cancel` parameters, no? this is the valid error on the screenshot

Reply:
Maybe I don't get something, we created a `CancelBatchFileParameters`, done `setFileUri()`, that is a necessary call, and now are unable to do `setReason()`?

Reply:
setFileUri return type hint must be `CancelBatchFileParameters `, not `this`.

Reply:
I mean I don't get how you get this error.

Reply:
Your ide thinks it's instance of `process batch something`

Reply:
> setFileUri return type hint must be `CancelBatchFileParameters `, not `this`.

It would be weird if `ProcessBatchActionParameters.setFileUri` would return `CancelBatchFileParameters`

Reply:
> setFileUri return type hint must be `CancelBatchFileParameters `, not `this`.

It literally does `return this;`

Reply:
> Your ide thinks it's instance of `process batch something`

That's not my IDE, that's tsc. Maybe it's because `ProcessBatchActionParameters.setFileUri` return type hint is `ProcessBatchActionParameters`

Reply:
you are right. disregard

Reply:
> That's not my IDE, that's tsc.

I built and ran tests - no this error. strange. but anyway, if it works then it works

Reply:
Looked at the jobs create/update params - they are defined with generics exactly because of this, so that return type typehint can be just whatever you pass as a generic (`T extends YourParentClass`) and in this case you can use methods from parent and from child without any issues you showed on screenshot above. just a comment, no actions point here.

Reply:
And it's better than a return typehint of `this` how?